### ISSUE TITLE
remove check for arity of python callbacks

### ISSUE THREAD

**Issue by noskill** [opencog/atomspace#1994]
There is method that checks that arity of callback in GroundedSchemaNode is the same as number of arguments defined in ExecutionOutputLink:
https://github.com/opencog/atomspace/blob/062469366854bc856dac9c914f030d6ad0461337/opencog/cython/PythonEval.cc#L702 It returns incorrect argument count for python functions with variable number of arguments, like

```
def test(a, b, *args):
      pass
```

It is possible correctly determine that argument count is variable with **inspect** module. On the other hand why not wait for error from python interpreter in case of argument count mismatch. That would cause RuntimeException to be thrown just like in case of argument count mismatch.
So should we remove this check?

**Comment by vsbogd** [comment]
I agree with @noskill suggestion, it is not clear why we are checking arity manually instead of delegating it to the Python interpreter.

**Comment by linas** [comment]
Yes, removing this seems like a reasonable suggestion, if you are certain that python is also performing this check.  Are you certain? If python is not performing this check, then removing this turns an error into a silent error. 

Silent errors are notoriously hard to debug: someone young and inexperienced, or maybe tired and not paying attention, accidentally passes two arguments to a function that expects only one. If no exception is thrown, everything seems fine, at first ... it can take hours to figure out why its not working as hoped. Silent errors are deadly.

It would be great if there was a unit test added, which pases one argument too many, and one too few, and checks for the exception.

**Comment by linas** [comment]
This was #2012
